### ISSUE TITLE
Cursor position change event not triggering if snapToSeries is active

### ISSUE THREAD

**Issue by andrija29** [amcharts/amcharts4#3387]
I have snapToSeries enabled for my series and I wanna have listener when cursor position changes. Problem is that `cursorpositionchanged` triggers just on the tick that I click on the chart, but there is slight animation that offsets the cursor while it is moving to snap to closest series point; `cursorpositionchanged` does not trigger again, which gives me wrong cursor position at the end of snap animation.

For example:
If my X point is at at X = 10 and next one at X = 15, when I click on the chart at X = 12 `cursorpositionchanged` triggers and cursor object at that tick is at 12, then cursor proceeds to move back to X = 10 (snap animation) but `cursorpositionchanged` does not trigger again and thus I am left with info that cursor is currently at X = 12.

I am happy to provide additional info if necessary.

**Comment by martynasma** [comment]
Are you referring to cursor's `point` property?

I can't seem to be able to get the behavior you described.

Maybe if you could post a CodePen example.

**Comment by andrija29** [comment]
@martynasma I am sorry for late response, I didn't saw notification for some reason.
I made this example so you can easily see what I am talking about. https://codepen.io/andrija29/pen/wvoZpmZ
When you click on chart (please use mobile agent so it's click instead of hover) and cursor position change event triggers, xPosition of the cursor is not the position where it ends up after chart snaps it to the closest point, it's rather point where user clicked.

So basically I cannot get real position of the cursor since xPosition at the time of click is coordinate at where the click event happened, and `cursorpositionchanged` event does not trigger again after chart moves cursor to snap on some value.

**Comment by andrija29** [comment]
It is definitely a bug since if I add a timeout of 100ms that starts when event triggers, I am getting correct coordinates. Problem is that you do not trigger `cursorpositionchanged` event when you snap cursor to value.

**Comment by martynasma** [comment]
Before anything else, there's one thing that needs to be fixed in your demo:

```
xAxes1.baseInterval = [{ timeUnit: "hour", count: 1 }];
```

`baseInterval` expects an object, not an array of objects:

```
xAxes1.baseInterval = { timeUnit: "hour", count: 1 };
```

As for the events, I did try on mobile emulator in Chrome. `cursorpositionchanged` does get triggered when cursor is snapping.

The only time it does not get called when snapping is when the snapping would occur to the same place the cursor is already snapped to.

I'm afraid we won't be able to change this behavior.

If you found a workaround using setTimeout, then I suggest using it.

**Comment by andrija29** [comment]
@martynasma I don't think that we understand each other here.
It does get triggered, but it gets triggered on initial click event at which (in that tick) coordinates are on the place click happened. Your code moves (snaps) cursor to the place and event does not get triggered second time. That means that I am left with coordinates from initial click, but cursor is in different position because you snapped it.

**Comment by martynasma** [comment]
I think we do :)

I mentioned the situations where `cursorpositionchanged` gets triggered on snap, and when not.

I also mentioned, that we won't be able to change current behavior. Sorry :(
